### DISCUSSION TITLE
What does `jobs.<job_id>.with.<with_id>` match?

### DISCUSSION THREAD

**Discussion by rhysd** [community/community#148153] [Actions] [answered]
### Select Topic Area

Question

### Body

The ['Context availability' table](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#context-availability) has `jobs.<job_id>.with.<with_id>` workflow key. And it is apparently different from `jobs.<job_id>.with`.

However I couldn't find what `jobs.<job_id>.with.<with_id>` actually matches. What does it match?

**Comment by Kik07L** [comment, marked as answer]
The jobs.<job_id>.with.<with_id> syntax you mentioned does not match any standard syntax in GitHub Actions workflows. The correct syntax for specifying inputs to a job or a step within a job is jobs.<job_id>.with.

Job-level inputs: GitHub Actions does not support with at the job level. Inputs are typically specified at the step level within a job.

Step-level inputs: The with keyword is used to pass inputs to an action or a script within a step.

Example of using with at the step level:


name: Example Workflow

on: [push]

jobs:
  example-job:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - name: Run a script
        run: echo "Hello, world!"

      - name: Example action
        uses: actions/example-action@v1
        with:
          input1: value1
          input2: value2
          
   In this example, the with keyword is used to pass input1 and input2 to actions/example-action@v1.

**Reply by rhysd** [reply]
Thank you for your answer. So the official document lists a meaningless workflow key, right?

**Reply by Kik07L** [reply]
the "jobs.<job_id>.with.<with_id>" structure is not standard in GitHub Actions' syntax. The documentation might be incorrectly i dont really know.

**Reply by rhysd** [reply]
OK, I'll suggest a fix for the document later. Thanks again.

**Comment by joshmgross** [comment]
It's an input for a reusable workflow job - https://docs.github.com/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idwithinput_id

https://docs.github.com/actions/sharing-automations/reusing-workflows
